cmd/istioctl/mixer.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	// TODO This should be github.com/googleapis/googleapis/google/rpc
+	rpc "github.com/googleapis/googleapis"
+
+	"github.com/ghodss/yaml"
+	"github.com/spf13/cobra"
+)
+
+// TODO This should come from something like istio.io/api instead of
+// being hand copied from istio.io/mixer.
+type mixerAPIResponse struct {
+	Data   interface{} `json:"data,omitempty"`
+	Status rpc.Status  `json:"status,omitempty"`
+}
+
+var (
+	mixerFile          string
+	mixerAPIServerAddr string
+
+	mixerCmd = &cobra.Command{
+		Use:          "mixer",
+		Short:        "Istio Mixer configuration",
+		SilenceUsage: true,
+		PersistentPreRunE: func(c *cobra.Command, args []string) error {
+			if mixerAPIServerAddr == "" {
+				return errors.New("no mixer API server specified (use --mixer)")
+			}
+			return nil
+		},
+	}
+
+	mixerRuleCmd = &cobra.Command{
+		Use:          "rule",
+		Short:        "Istio Mixer Rule configuration",
+		SilenceUsage: true,
+	}
+
+	mixerRuleCreateCmd = &cobra.Command{
+		Use:   "create <scope> <subject>",
+		Short: "Create an Istio Mixer rule",
```

Comment:
Create a Istio Mixer rules document.

This document has multiple rules within in.